lib/ts/recipe/session/sessionAuth.tsx

```diff
@@ -109,28 +168,35 @@ const SessionAuth: React.FC<PropsWithChildren<Props>> = ({ children, ...props })
         function onHandleEvent(event: RecipeEventWithSessionContext) {
             switch (event.action) {
                 case "SESSION_CREATED":
-                    setContext(event.sessionContext);
+                    setContextUpdate(event.sessionContext);
                     return;
                 case "REFRESH_SESSION":
-                    setContext(event.sessionContext);
+                    setContextUpdate(event.sessionContext);
                     return;
                 case "ACCESS_TOKEN_PAYLOAD_UPDATED":
-                    setContext(event.sessionContext);
+                    setContextUpdate(event.sessionContext);
+                    return;
+                case "API_INVALID_CLAIM":
+                    // In general the user should not be calling APIs that throw this
+                    // so this may suggest that a claim was invalidated
```

Comment:
Why not?

Reply:
It's not really good practice to make calls (and show buttons) that you know will fail. 
We may not have enough information (so the error only appears on the server-side), which should be rare or the information we have is outdated so we should re-validate the claims.